src/ensembl/src/content/app/custom-download/components/checkbox-with-selects/CheckboxWithSelects.test.tsx

```diff
@@ -31,107 +32,117 @@ const createOption = (isSelected = false) => ({
 
 const onChange = jest.fn();
 
+const defaultProps = {
+  onChange: onChange,
+  label: 'foo',
+  selectedOptions: [],
+  options: times(5, () => createOption())
+};
+
 describe('<CheckboxWithSelects />', () => {
+  const renderCheckboxWithSelects = (
+    props?: Partial<CheckboxWithSelectsProps>
+  ) => render(<CheckboxWithSelects {...defaultProps} {...props} />);
+
   afterEach(() => {
     jest.resetAllMocks();
   });
 
-  let wrapper: any;
-  const defaultProps = {
-    onChange: onChange,
-    label: 'foo',
-    selectedOptions: [],
-    options: times(5, () => createOption())
-  };
-
   it('renders without error', () => {
-    wrapper = mount(<CheckboxWithSelects {...defaultProps} />);
-    expect(wrapper.find(CheckboxWithSelects).length).toEqual(1);
+    expect(() => {
+      renderCheckboxWithSelects();
+    }).not.toThrow();
   });
 
   it('does not check the checkbox when there are no options selected', () => {
-    wrapper = mount(<CheckboxWithSelects {...defaultProps} />);
+    const { container } = renderCheckboxWithSelects();
 
-    expect(wrapper.find(Checkbox).prop('checked')).toBe(false);
+    expect(container.querySelector('.checkboxHolder .checked')).toBeFalsy();
   });
 
   it('displays one Select when the checkbox is unchecked', () => {
-    wrapper = mount(<CheckboxWithSelects {...defaultProps} />);
+    const { container } = renderCheckboxWithSelects();
+
+    expect(container.querySelector('.checkboxHolder .checked')).toBeFalsy();
 
-    expect(wrapper.find(Checkbox).prop('checked')).toBe(false);
-    expect(wrapper.find(Select).length).toBe(1);
+    expect(container.querySelectorAll('.select').length).toBe(1);
   });
 
   it('displays one Select when the checkbox is checked', () => {
-    wrapper = mount(<CheckboxWithSelects {...defaultProps} />);
+    const { container } = renderCheckboxWithSelects();
 
-    wrapper.find(Checkbox).find('.defaultCheckbox').simulate('click');
-    expect(wrapper.find(Select).length).toBe(1);
+    const checkboxElement = container.querySelector('.defaultCheckbox');
+
+    userEvent.click(checkboxElement as HTMLElement);
+
+    expect(container.querySelectorAll('.select').length).toBe(1);
   });
 
   it('automatically checks the checkbox if at least one option is selected', () => {
-    wrapper = mount(
-      <CheckboxWithSelects {...defaultProps} selectedOptions={['one']} />
-    );
-
-    expect(wrapper.find(Checkbox).prop('checked')).toBe(true);
+    const { container } = renderCheckboxWithSelects({
+      selectedOptions: [defaultProps.options[0].value]
+    });
+    expect(container.querySelector('.checkboxHolder .checked')).toBeTruthy();
```

Comment:
We should probably find the first checkbox here and make sure that it is checked?

Reply:
That's done